velox/functions/prestosql/StringFunctions.h

```diff
@@ -507,4 +507,76 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+template <typename T>
+struct FromBigEndian32 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void callNullFree(
+      out_type<int32_t>& result,
+      const null_free_arg_type<Varbinary>& input) {
+    static const auto kTypeLength = sizeof(int32_t);
+    VELOX_USER_CHECK(
+        input.size() == kTypeLength,
+        "Expected 4-byte input, but got instead: {}",
+        input.size());
+    auto buffer = input.data();
+    result = ((int32_t)buffer[0] & 0xff) << 24 |
+        ((int32_t)buffer[1] & 0xff) << 16 | ((int32_t)buffer[2] & 0xff) << 8 |
+        ((int32_t)buffer[3] & 0xff);
+  }
+};
+
+template <typename T>
+struct ToBigEndian32 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void callNullFree(
+      out_type<Varbinary>& result,
+      const null_free_arg_type<int32_t>& input) {
+    static const auto kTypeLength = sizeof(int32_t);
+    auto value = folly::Endian::big(input);
+    result.resize(kTypeLength);
+    std::memcpy(result.data(), &value, kTypeLength);
+  }
+};
+
+template <typename T>
+struct FromBigEndian64 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void callNullFree(
+      out_type<int64_t>& result,
+      const null_free_arg_type<Varbinary>& input) {
+    static const auto kTypeLength = sizeof(int64_t);
+    VELOX_USER_CHECK(
+        input.size() == kTypeLength,
+        "Expected 8-byte input, but got instead: {}",
+        input.size());
+    auto buffer = input.data();
+    result = ((int64_t)buffer[0] & 0xff) << 56 |
+        ((int64_t)buffer[1] & 0xff) << 48 | ((int64_t)buffer[2] & 0xff) << 40 |
+        ((int64_t)buffer[3] & 0xff) << 32 | ((int64_t)buffer[4] & 0xff) << 24 |
+        ((int64_t)buffer[5] & 0xff) << 16 | ((int64_t)buffer[6] & 0xff) << 8 |
+        ((int64_t)buffer[7] & 0xff);
+  }
+};
+
+template <typename T>
+struct ToBigEndian64 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void callNullFree(
+      out_type<Varbinary>& result,
+      const null_free_arg_type<int64_t>& input) {
+    static const auto kTypeLength = sizeof(int64_t);
```

Comment:
From doc here: https://en.cppreference.com/w/cpp/language/sizeof
sizeof is a constant expression. Do we need a variable to store the value.